src/main/java/com/ldtteam/structurize/component/ModDataComponents.java

```diff
@@ -0,0 +1,39 @@
+package com.ldtteam.structurize.component;
+
+import com.ldtteam.structurize.api.constants.Constants;
+import com.ldtteam.structurize.items.AbstractItemWithPosSelector.PosSelection;
+import com.ldtteam.structurize.items.ItemTagTool.TagData;
+import com.ldtteam.structurize.util.ScanToolData;
+import com.mojang.serialization.Codec;
+import net.minecraft.core.component.DataComponentType;
+import net.minecraft.network.RegistryFriendlyByteBuf;
+import net.minecraft.network.codec.StreamCodec;
+import net.neoforged.neoforge.registries.DeferredHolder;
+import net.neoforged.neoforge.registries.DeferredRegister;
+
+public class ModDataComponents
+{
+    public static final DeferredRegister.DataComponents REGISTRY = DeferredRegister.createDataComponents(Constants.MOD_ID);
+
+    public static final DeferredHolder<DataComponentType<?>, DataComponentType<PosSelection>> POS_SELECTION =
+        savedSynced("pos_selection", PosSelection.CODEC, PosSelection.STREAM_CODEC);
+    public static final DeferredHolder<DataComponentType<?>, DataComponentType<TagData>> TAGS_DATA =
+        savedSynced("tags", TagData.CODEC, TagData.STREAM_CODEC);
```

Comment:
Can't we just make that nice Function blocks instead of stream codecs? Those are much easier to debug and log trace errors with

Reply:
the points is you dont debug codecs :D stream_codecs are rewritable to functions because they're essentially only functions wrappers now, dfu codecs not however